### PR TITLE
Support time.Time

### PR DESCRIPTION
**Pull request by mcdee**
This allows encoding and decoding of the `time.Time` type in structs, to `uint64` values.

### CODE REVIEW

File: tests/time.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package tests
```
Comment (ferranbt) [inline tests/time.go:1]: Please write these tests in [testcases](https://github.com/ferranbt/fastssz/tree/master/sszgen/testcases) instead.

File: tests/time_encoding.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// UnmarshalSSZ ssz unmarshals the TimeType object
+func (t *TimeType) UnmarshalSSZ(buf []byte) error {
+	var err error
+	size := uint64(len(buf))
+	if size != 16 {
+		return ssz.ErrSize
+	}
+
+	// Field (0) 'Timestamp'
+	t.Timestamp = time.Unix(int64(ssz.UnmarshallUint64(buf[0:8])), 0).UTC()
```
Comment (ferranbt) [inline tests/time_encoding.go:36]: You need to add an entry [here](https://github.com/ferranbt/fastssz/blob/bd3f11aa9eaa37e3995d5eaf80ea994dc9ae20c2/sszgen/generator/generator.go#L454) to resolve the import.

**Review by ferranbt — CHANGES_REQUESTED** [review]
It looks good. Only minor changes to follow the same standard for the generated code.

File: tests/time.go (hunk 1 above)
Comment (mcdee) [reply tests/time.go]: Moved.

File: tests/time_encoding.go (hunk 2 above)
Comment (mcdee) [reply tests/time_encoding.go]: Added.

File: sszgen/generator/unmarshal.go [hunk 3]
```diff
@@ -104,6 +104,9 @@ func (v *Value) unmarshal(dst string) string {
 	case TypeBool:
 		return fmt.Sprintf("::.%s = ssz.UnmarshalBool(%s)", v.name, dst)
 
+	case TypeTime:
+		return fmt.Sprintf("::.%s = time.Unix(int64(ssz.UnmarshallUint64(%s)), 0).UTC()", v.name, dst)
```
[the remark this answers is not in the document]
Comment (ferranbt) [reply sszgen/generator/unmarshal.go:108]: Sorry, you do not need to do [this](https://github.com/ferranbt/fastssz/pull/100#discussion_r935317133) after this change.

[the remark this answers is not in the document]
Comment (mcdee) [reply sszgen/generator/unmarshal.go]: Should be addressed, but please take a look to confirm.

(Note that this removes the requirement for the time import in the earlier comment).

File: encode.go [hunk 4]
```diff
@@ -108,6 +114,14 @@ func MarshalBool(dst []byte, b bool) []byte {
 	return dst
 }
 
+// MarshalTime marshals a time to dst
+func MarshalTime(dst []byte, t time.Time) []byte {
+	buf := make([]byte, 8)
```
Comment (ferranbt) [inline encode.go:119]: Please do:
```
func MarshalTime(dst []byte, t time.Time) []byte {
   return MarshalUint64(dst, t.Unix())
}
```

File: encode.go [hunk 5]
```diff
@@ -66,6 +67,11 @@ func UnmarshalBool(src []byte) bool {
 	return false
 }
 
+// UnmarshalTime unmarshals a time.Time from the src input
+func UnmarshalTime(src []byte) time.Time {
+	return time.Unix(int64(binary.LittleEndian.Uint64(src)), 0).UTC()
```
Comment (ferranbt) [inline encode.go:72]: Please do:
```
func UnmarshalTime(src []byte) time.Time {
   return time.Unix(int64(UnmarshallUint64(src)), 0).UTC()}
```

File: sszgen/testcases/time.go [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package testcases
+
+import "time"
+
```
Comment (ferranbt) [inline sszgen/testcases/time.go:4]: Add a `go:generate` expression like [this](https://github.com/ferranbt/fastssz/blob/master/sszgen/testcases/case1.go#L3) on the top to generate these test files automatically.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: encode.go (hunk 4 above)
Comment (mcdee) [reply encode.go]: Updated.

**Review by ferranbt — APPROVED** [review]

**Pull request merged**